CRM/Core/Payment.php

```diff
@@ -1629,7 +1965,10 @@ public function subscriptionURL($entityID = NULL, $entity = NULL, $action = 'can
    *
    * @return string
    */
-  protected function getPaymentDescription($params, $length = 24) {
+  protected function getPaymentDescription($params = [], $length = 24) {
+    if ($this->getDescription()) {
+      return $this->getDescription();
+    }
```

Comment:
Not sure what is best here - it seems a shame to override the useful getPaymentDescription function completely instead of just using getDescription instead of `$params['description']`.

Reply:
OK - I've altered this to move the string logic to getDescription